Detectors/ITSMFT/ITS/reconstruction/include/ITSReconstruction/CookedTrackerTask.h

```diff
@@ -43,22 +44,23 @@ class CookedTrackerTask : public FairTask
 
   void setContinuousMode(bool mode) { mContinuousMode = mode; }
   bool getContinuousMode() { return mContinuousMode; }
-  
+
+  TrivialVertexer& getVertexer() { return mVertexer; }
+
```

Comment:
Do we really need to hard-code the vertexer grabbing the MC-truth vertex into the CookedTrackerTask? I think the vertex (real or idealized one) should be reconstructed in a separated component and taken by the tracker as an input. Hardcoding idealized vertexer into the tracking we will get problem with using real external vertex. In any case, even if we want the ITS vertex to serve just to this reconstruction, I think it is better to do this via generic vertexer plugin, from which then the trivial vertexer  (or others) can be derived.

Reply:
@shahor02 Sure.  A better interface is coming.